ccmbioinfo/lfs
# LFS-121: 'Answer' widget that implements the common functionality of different answer types

To test, create the following files:
`modules/data-entry/src/main/resources/SLING-INF/content/Extensions.json`
```
{
    "jcr:primaryType": "sling:Folder",
    "TestRig": {
        "jcr:primaryType": "lfs:Extension",
        "lfs:extensionPointId": "lfs/coreUI/sidebar/entry",
        "lfs:extensionName": "Questionnaire test",
        "lfs:defaultOrder": 0,
        "lfs:targetURL": "Questionnaires",
        "lfs:defaultDisabled": false,
        "lfs:icon": "asset:lfs-dataentry.questionnaireIcon.js",
        "lfs:extensionRenderURL": "asset:lfs-dataentry.testRig.js",
        "jcr:content": {
            "jcr:primaryType": "nt:resource",
            "jcr:data": "",
            "jcr:mimeType": "text/html"
        }
    }
}
```

`modules/data-entry/src/main/frontend/src/questionnaire/testRig.jsx`
```
//
//  Licensed to the Apache Software Foundation (ASF) under one
//  or more contributor license agreements.  See the NOTICE file
//  distributed with this work for additional information
//  regarding copyright ownership.  The ASF licenses this file
//  to you under the Apache License, Version 2.0 (the
//  "License"); you may not use this file except in compliance
//  with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing,
//  software distributed under the License is distributed on an
//  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
//  KIND, either express or implied.  See the License for the
//  specific language governing permissions and limitations
//  under the License.
//

import React from "react";

import MultipleChoice from "./MultipleChoice";
import Question from "./Question";

export default function TestRig(props) {
  return (
    <React.Fragment>
      <Question
        text="Test multiple choice"
        description="No input"
        >
        <MultipleChoice
          max={1}
          defaults={[
            {"id": "1", "label": "1"},
            {"id": "2", "label": "2"},
            {"id": "3", "label": "3"}
          ]}
          />
      </Question>
      <Question
        text="Test multiple choice"
        description="Text input"
        >
        <MultipleChoice
          max={1}
          defaults={[
            {"id": "1", "label": "1"},
            {"id": "2", "label": "2"},
            {"id": "3", "label": "3"}
          ]}
          text="Test multiple choice"
          description="Text input"
          input
          />
      </Question>
    </React.Fragment>);
}
```

Also alter the 'entry' (line 8) value of 
`modules/data-entry/src/main/frontend/webpack.config.js`:
```
  entry: {
    [module_name + 'redirect']: './src/dataQuery/redirect.js',
    [module_name + 'showQuery']: './src/dataQuery/query.js',
    [module_name + 'multipleChoice']: './src/questionnaire/MultipleChoice.jsx',
    [module_name + 'answer']: './src/questionnaire/Answer.jsx'
    [module_name + 'testRig']: './src/questionnaire/testRig.jsx',
    [module_name + 'questionnaireIcon']: '@material-ui/icons/Assignment.js'
  },
```
